src/components/progress-bar/progress-bar.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    Attribute,
+    Component,
+    ChangeDetectionStrategy,
+    ElementRef,
+    HostBinding,
+    Input,
+    ViewEncapsulation,
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+// TODO(josephperrott): Add ARIA attributes for progressbar "for".
+
+
+/** Display modes of Progress Bar*/
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+  @CONST() static BUFFER = 'buffer';
+  @CONST() static QUERY = 'query';
+}
+
+
+/**
+ * <md-progress-bar> component.
+ */
+@Component({
+  selector: 'md-progress-bar',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-bar/progress-bar.html',
+  styleUrls: ['./components/progress-bar/progress-bar.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdProgressBar {
+  /**
+   * Value of the progress bar.
+   *
+   * Input:number, defaults to 0.
+   * value_ is bound to the host as the attribute aria-valuenow.
+   */
+  @Input('value')
+  value_: number = 0;
+  @HostBinding('attr.aria-valuenow')
+  get _value() {
+    return this.value_;
+  }
```

Comment:
Since we want to bind `value` input to `value_`, we need to keep `@Input` as a decorator on the `value_` definition.